src/main/java/ch/jalu/configme/properties/ArrayProperty.java

```diff
@@ -0,0 +1,127 @@
+package ch.jalu.configme.properties;
+
+import ch.jalu.configme.properties.types.BeanPropertyType;
+import ch.jalu.configme.properties.types.PropertyType;
+import ch.jalu.configme.resource.PropertyReader;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class ArrayProperty<T> extends BaseProperty<T[]> {
+
+    private final PropertyType<T> type;
+
+    /**
+     * Constructor.
+     *
+     * @param path         the path of the property
+     * @param defaultValue the default value of the property
+     * @param type         the property type
+     */
+    public ArrayProperty(String path, T[] defaultValue, PropertyType<T> type) {
+        super(path, defaultValue);
+
+        if (type instanceof BeanPropertyType<?>) {
+            throw new IllegalArgumentException("BeanPropertyType not support for array property (maybe, temporarily)");
```

Comment:
"Is it because BeanPropertyType#convert is not supported?" - Yes,I could not come up with the implementation of the conversion, and I hope this can be done by you

Reply:
OK, after I do the changes to Mapper we should be good to go :)